ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/properties/BackgroundStyle.kt

```diff
@@ -0,0 +1,134 @@
+package com.revenuecat.purchases.ui.revenuecatui.components.properties
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.requiredSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.revenuecat.purchases.paywalls.components.common.Background
+import com.revenuecat.purchases.paywalls.components.properties.ColorInfo
+import com.revenuecat.purchases.paywalls.components.properties.ColorScheme
+import com.revenuecat.purchases.ui.revenuecatui.components.ktx.urlsForCurrentTheme
+import com.revenuecat.purchases.ui.revenuecatui.components.modifier.background
+
+/**
+ * Ready to use background properties for the current theme.
+ */
+internal sealed interface BackgroundStyle {
+    @JvmInline
+    value class Color(@JvmSynthetic val color: ColorStyle) : BackgroundStyle
+
+    @JvmInline
+    value class Image(@JvmSynthetic val painter: Painter) : BackgroundStyle
+}
+
+@JvmSynthetic
+@Composable
+internal fun Background.toBackgroundStyle(): BackgroundStyle =
+    when (this) {
+        is Background.Color -> BackgroundStyle.Color(color = value.toColorStyle())
+        is Background.Image -> {
+            val imageUrls = value.urlsForCurrentTheme
+            BackgroundStyle.Image(
+                painter = rememberAsyncImagePainter(
+                    model = imageUrls.webp,
+                    placeholder = rememberAsyncImagePainter(
+                        model = imageUrls.webpLowRes,
+                        error = null,
+                        fallback = null,
+                        contentScale = ContentScale.Crop,
+                    ),
+                    error = null,
+                    fallback = null,
+                    contentScale = ContentScale.Crop,
+                ),
+            )
```

Comment:
I'm using the low-res variant as a placeholder for now. We can think about supporting [blurhash](https://blurha.sh/), but ideally the images are cached already before the paywall is shown.

Reply:
Hmm I'm wondering if we should have a bundled placeholder (or background basically 😅. But yeah hopefully this will be enough.

Reply:
Yea indeed, something to think about!